tests/test_workspace_runtime.py

```
@@ -2646,6 +2646,89 @@ def assert_registry_encloses_workspace() -> None:
+    contender_failures: list[BaseException] = []
+
+    def contend() -> None:
+        try:
+            contender = (
+                registry.exclusive_lock()
+                if lock_name == "registry"
+                else leases.workspace_lock(REPO_UUID)
+            )
+            with contender:
+                contender_acquired.set()
+        except BaseException as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `BaseException`

(BLE001)